client/src/mapper-consents/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { apiErrorMessage, getMapperConsents, useApi } from '../api';
+import { BeatmapInline } from '../BeatmapInline';
+import { IMapperConsent } from '../interfaces';
+import { UserInline } from '../UserInline';
+
+function consentToCell(consent?: 0 | 1 | 2 | boolean) {
+  let index: 'null' | 0 | 1 | 2;
+  if (consent === true) {
+    index = 1
+  } else if (consent === false) {
+    index = 0
+  } else {
+    index = consent == null ? 'null' : consent;
+  }
+  const mapping = {
+    'null': ['no reply', 'pending'],
+    0: ['no', 'error'],
+    1: ['yes', 'success'],
+    2: ['unreachable', 'pending']
+  };
+  const [consentStr, className] = mapping[index];
+
+  return (
+    <td className={className}>
+      {consentStr}
+    </td>
+  )
+}
+
+function maybeBeatmapsetTable(consent: IMapperConsent) {
+  if (consent.beatmapset_consents.length === 0) {
+    return
+  }
+  return (
```

Comment:
more js nitpick, this simplifies to `consent.beatmapset_consents.length > 0 && ...` because `false` is one of react's reserved values to mean "don't render me"

Reply:
also i dont think this needs to be a separate function/component